src/assignment-logger.spec.ts

```diff
@@ -11,6 +12,22 @@ describe('IAssignmentEvent', () => {
       timestamp: new Date().toISOString(),
       subjectAttributes: { age: 25, country: 'USA' },
       holdoutKey: 'holdout_key_123',
+      details: {
```

Comment:
The purpose of this test was not obvious to me until I ran it myself. It will fail if any required attribute in `IAssignmentEvent` is missing but will pass if attributes not specified in `IAssignmentEvent` are included in the `event` object (the latter behavior is described by the test "it" statement)

Just a suggestion (feel free to take it or leave it): if it's not too much trouble, you could add a test that checks for a thrown exception if `details` (a required attribute in the `event` object) is missing